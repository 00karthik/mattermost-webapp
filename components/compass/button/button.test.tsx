// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Button from './button';

describe('components/compass/Button', () => {
    test('should match snapshot', () => {
        const Props = {
            style: 'primary',
            size: 'small',
            label: 'Hello',
        };

        const wrapper = shallow(
            <Button
                {...Props}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('button').props().className).toBe('Button Button___primary Button___small');
    });
});
