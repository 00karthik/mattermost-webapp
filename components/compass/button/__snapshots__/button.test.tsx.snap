// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/compass/Button should match snapshot 1`] = `
<div
  className="Button__wrapper"
>
  <button
    aria-label="button"
    className="Button Button___primary Button___small"
    tabIndex={-1}
  >
    <span
      className="Label Label___standard Button_label"
    >
      Hello
    </span>
  </button>
</div>
`;
